narwhals/functions.py

```diff
@@ -406,6 +411,100 @@ def _from_dict_impl(
     return from_native(native_frame, eager_only=True)
 
 
+def from_pycapsule(
```

Comment:
Personally I'd call this `from_arrow` and under the hood use the PyCapsule Interface. I think the PyCapsule itself is an implementation detail, and not something the user should need to know about.

Reply:
based on https://github.com/pola-rs/polars/issues/12530#issuecomment-2419570358, i think this risks being confused with Polars' `from_arrow`, which does something different

i'd be more inclined to use `from_pycapsule` then

Reply:
Do you only have top-level functions? Do you export a `DataFrame`-like class? Why not put it on that class?

But there is a difference between the two because your function seems to only convert to a `DataFrame` while the polars function tries to decide whether it should return a `Series` or `DataFrame`.

`from_pycapsule` still doesn't really resolve the ambiguity, but you could change it to `from_arrow_dataframe` or `from_arrow_df` or `from_arrow_table`.

Reply:
thanks! yeah maybe class constructors are the way to go for this one